src/window/concept_frame.rs

```diff
@@ -639,22 +640,20 @@ impl Frame for ConceptFrame {
                             // If theres no stored font data, find the first ttf regular sans font and
                             // store it
                             if self.font_data.is_none() {
-                                if let Some(font) = fontconfig::FontConfig::new()
-                                    .unwrap()
-                                    .get_regular_family_fonts(&font_face)
-                                    .unwrap()
-                                    .iter()
-                                    .find(|p| p.extension().map(|e| e == "ttf").unwrap_or(false))
-                                {
-                                    let mut font_data = Vec::new();
-                                    if let Ok(mut file) = ::std::fs::File::open(font) {
-                                        match file.read_to_end(&mut font_data) {
-                                            Ok(_) => self.font_data = Some(font_data),
-                                            Err(err) => {
-                                                log::error!("Could not read font file: {}", err)
-                                            }
-                                        }
-                                    }
+                                let font_config = fontconfig::FontConfig::new().ok();
+                                let fonts = font_config.and_then(|font_config| {
+                                    font_config.get_regular_family_fonts(&font_face).ok()
+                                });
+                                let font = fonts.and_then(|regular_family_fonts| {
+                                    regular_family_fonts
+                                        .iter()
+                                        .cloned()
+                                        .find(|p| p.extension().map_or(false, |e| e == "ttf"))
+                                });
+                                let font_bytes = font.and_then(|font| std::fs::read(font).ok());
+                                match font_bytes {
+                                    Some(bytes) => self.font_data = Some(bytes),
+                                    None => error!("No font could be found"),
```

Comment:
I think we should store in `self.font_data` something indicating that the font could not be loaded if that was the case, for example by making the field an `Option<Result<Vec<u8>, ()>>`. This way, `None` indicates that the search has not been done yet, and `Some(Err(()))` indicates that it has failed.

Otherwise, SCTK will attempt again to load the font every time a redraw is triggered, which seems unnecessary.